julia-package/brainflow/src/board_shim.jl

```diff
@@ -38,620 +36,172 @@ end
-    else
-        ec = ccall((:get_other_channels, "libBoardController.so"), Cint, (Cint, Ptr{Cint}, Ptr{Cint}), Int32(board_id), channels, len)
-    end
-    if ec != Integer(STATUS_OK)
-        throw(BrainFlowError(string("Error in get board info ", ec), ec))
-    end
-    # julia counts from 1
-    value = channels[1:len[1]] .+ 1
-    value
-end
-
-
-function get_temperature_channels(board_id::AnyIntType)
-    channels = Vector{Cint}(undef, 512)
-    len = Vector{Cint}(undef, 1)
-    ec = STATUS_OK
-    # due to this bug https://github.com/JuliaLang/julia/issues/29602 libname should be hardcoded
-    if Sys.iswindows()
-        ec = ccall((:get_temperature_channels, "BoardController.dll"), Cint, (Cint, Ptr{Cint}, Ptr{Cint}), Int32(board_id), channels, len)
-    elseif Sys.isapple()
-        ec = ccall((:get_temperature_channels, "libBoardController.dylib"), Cint, (Cint, Ptr{Cint}, Ptr{Cint}), Int32(board_id), channels, len)
-    else
-        ec = ccall((:get_temperature_channels, "libBoardController.so"), Cint, (Cint, Ptr{Cint}, Ptr{Cint}), Int32(board_id), channels, len)
-    end
-    if ec != Integer(STATUS_OK)
-        throw(BrainFlowError(string("Error in get board info ", ec), ec))
-    end
-    # julia counts from 1
-    value = channels[1:len[1]] .+ 1
-    value
-end
-
-
-function get_resistance_channels(board_id::AnyIntType)
-    channels = Vector{Cint}(undef, 512)
-    len = Vector{Cint}(undef, 1)
-    ec = STATUS_OK
-    # due to this bug https://github.com/JuliaLang/julia/issues/29602 libname should be hardcoded
-    if Sys.iswindows()
-        ec = ccall((:get_resistance_channels, "BoardController.dll"), Cint, (Cint, Ptr{Cint}, Ptr{Cint}), Int32(board_id), channels, len)
-    elseif Sys.isapple()
-        ec = ccall((:get_resistance_channels, "libBoardController.dylib"), Cint, (Cint, Ptr{Cint}, Ptr{Cint}), Int32(board_id), channels, len)
-    else
-        ec = ccall((:get_resistance_channels, "libBoardController.so"), Cint, (Cint, Ptr{Cint}, Ptr{Cint}), Int32(board_id), channels, len)
-    end
-    if ec != Integer(STATUS_OK)
-        throw(BrainFlowError(string("Error in get board info ", ec), ec))
-    end
-    # julia counts from 1
-    value = channels[1:len[1]] .+ 1
-    value
-end
-
-
 struct BoardShim
 
     master_board_id::Int32
     board_id::Int32
     input_json::String
 
-    function BoardShim(id::AnyIntType, params::BrainFlowInputParams)
+    function BoardShim(id::Integer, params::BrainFlowInputParams)
         master_id = id
         if id == Integer(STREAMING_BOARD) || id == Integer(PLAYBACK_FILE_BOARD)
             try
                 master_id = parse(Int, id)
             catch
-                throw(BrainFlowError("you need to provide master board id to other_info field"))
+                throw(BrainFlowError("you need to provide master board id to other_info field", 1))
             end
         end
```

Comment:
I dont understand the purpose of this part. And I think it would even fail:
```
julia> parse(Int, Int32(5))
ERROR: MethodError: no method matching parse(::Type{Int64}, ::Int32)
````

Reply:
BrainFlowError Defined like
```
struct BrainFlowError <: Exception
    msg::String
    ec::Int32
end
```
First arg is a string and second one is one of exit codes, I think instead 1 need to use INVALID_ARGUMENTS_ERROR from https://github.com/brainflow-dev/brainflow/blob/master/julia-package/brainflow/src/exit_codes.jl#L16

Idea is to throw an exception of known type(BrainFlowError)

Reply:
it parses other_info field of BrainFlowInputParams and its a string, for some boards you need to provide int number to this string field.

Reply:
Yes I get the exception. I mean the purpose of the `if` statement at all.

Input `id` is an `Integer`, so no need to try to parse it again into an `Int` right?
```
function BoardShim(id::Integer, params::BrainFlowInputParams)
        master_id = id
        if id == Integer(STREAMING_BOARD) || id == Integer(PLAYBACK_FILE_BOARD)
            try
                master_id = parse(Int, id)
            catch
                throw(BrainFlowError("you need to provide master board id to other_info field", 1))
            end
        end
        new(master_id, id, JSON.json(params))
    end
```

Can we just remove the `if` and copy the `id` into the `master_id`:
```
function BoardShim(id::Integer, params::BrainFlowInputParams)
    new(id, id, JSON.json(params))
end
```
Or maybe add another constructor for the String. This should already throw a good exception if the string cannot be converted to an int.
```
function BoardShim(master_id::AbstractString, id::Integer, params::BrainFlowInputParams)
    int_master_id = parse(Int, master_id)
    new(int_master_id , id, JSON.json(params))
end
```

Reply:
Are there examples from these special boards, and how they should use the input_json generated from the BrainFlowInputParams?

Reply:
Oh, seems like I made a mistake. It should be like
```
 if id == Integer(STREAMING_BOARD) || id == Integer(PLAYBACK_FILE_BOARD)
            try
                master_id = parse(Int, BrainFlowInputParams.other_info)
            catch
                throw(BrainFlowError("you need to provide master board id to other_info field", 1))
            end
        end
```

Reply:
fixed